syntax = "proto2";
option optimize_for = LITE_RUNTIME;

import "common.proto";
package orwell.messages;

// Notify the state the robot is in (sent while playing)
message ServerRobotState {
	repeated Colour colour = 1;
	repeated Rfid rfid = 2;
	optional Ultrasound ultrasound = 3;
	optional Battery battery = 4;
}

enum Status {
	ON = 1;
	OFF = 2;
}

// Robot is detecting a Colour flag
message Colour {
	required uint64 timestamp = 1;
	required Status status = 2;
	required int32 colour = 3;
}

// Robot is detecting a RFID flag
message Rfid {
	required uint64 timestamp = 1;
	required Status status = 2;
	required string rfid = 3;
}

// Ask the server to register a new robot. The video address and port
// are provided.
// Answered by: Registered
message Register {
	// this temporary robot ID is only used when registering because
	// we do not know yet what ID the server will assign to the robot
	required string temporary_robot_id = 1;
	optional string video_url = 2;
	required string image = 3;
}

// Reply to a ping message to mesure timings
message Pong {
	repeated Timing timing = 1;
}
